packages/next/src/storage.ts

```diff
@@ -0,0 +1,29 @@
+import { Storage, StorageKey } from '@logto/node';
+
+import { NextRequestWithIronSession } from './types';
+
+export default class NextStorage implements Storage {
+  constructor(private readonly request: NextRequestWithIronSession) {}
+
+  async setItem(key: StorageKey, value: string) {
+    this.request.session[key] = value;
+  }
+
+  getItem(key: StorageKey) {
+    const value = this.request.session[key];
+
+    if (typeof value === 'string') {
+      return value;
+    }
+
+    return null;
+  }
+
+  removeItem(key: StorageKey) {
+    this.request.session[key] = undefined;
+  }
+
+  async save() {
```

Comment:
will it be better to call `save()` inside the functions above?

Reply:
That will need to change `setItem` to `async`, per offline discussion, consider refactor it later.